README.md

```diff
@@ -9,21 +9,72 @@ rustup target add riscv64gc-unknown-none-elf
 cargo build
 ```
 
-## running
+## Running
 
-There is a `Makefile` provided with targets for running various hosts in qemu.
+There is a `Makefile` provided with targets for running various hosts in QEMU.
 
 ### Prerequisites
 
-- rust toolchain from [rustup](rustup.rs)
-- rustup target add riscv64gc-unknown-none-elf
-- install `gcc-riscv64-unknown-elf` for `riscv64-unknown-elf-objcopy`
-- qemu-system-riscv64 - build using  qemu
-  [instructions](https://wiki.qemu.org/Hosts/Linux) with
+Toolchains:
+- Rust: Install [rustup](rustup.rs), then: `rustup target add riscv64gc-unknown-none-elf`
+- GCC: Install `gcc-riscv64-unknown-elf`
+
+QEMU:
+- Out-of-tree patches are required; see table below.
+- Build using QEMU [instructions](https://wiki.qemu.org/Hosts/Linux) with
   `--target-list=riscv64-softmmu`
-  - Note qemu 7.0.0 has a bug emulating riscv and a newer version from git must be used.
 
-## Test VM
+OpenSBI:
+- Use source from upstream: https://github.com/riscv-software-src/opensbi
+- Build: `PLATFORM=generic CROSS_COMPILE=riscv64-unknown-linux-gnu- make`
+
+Linux kernel (for booting Linux VMs):
+- Out-of-tree patches are required; see table below.
+- Build: `ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- make defconfig Image`
+
+Latest known-working branches:
+
+| Project | Branch |
+| ------- | ------ |
+| QEMU    | https://github.com/rivosinc/qemu/tree/salus-integration-08172022 |
+| Linux   | https://github.com/rivosinc/linux/tree/salus-integration-08172022 |
+| OpenSBI | use upstream: https://github.com/riscv-software-src/opensbi |
+
+### Linux VM
+
+The `make run_linux` target will boot a bare Linux kernel as the host VM,
+though it doesn't do much without any filesystems or emulated devices.
+
+To launch a more functional Debian VM:
+- Grab pre-built `riscv64-virt` images here: https://people.debian.org/~gio/dqib/
+- Launch QEMU with:
+  ```
+  qemu-system-riscv64 \
+   -M virt,aia=aplic-imsic,aia-guests=4 \
+   -cpu rv64,x-aia=true -m 2048 -smp "${NCPU}" \
+   -nographic \
+   -bios "${OPENSBI}" \
+   -kernel "${SALUS}" -device guest-loader,addr=0xc0200000,kernel="${KERNEL}" \
+   -append "console=hvc0 earlycon=sbi keep_bootcon root=LABEL=rootfs" \
+   -device guest-loader,addr=0xc2200000,initrd="${INITRD}" \
```

Comment:
I think it would be better to describe the relative path environment variables used here. It's pretty apparent for anybody who is used to the system. But newbies may find it bit difficult at first.

Reply:
I tried to simplify this in the Makefile and described the variables in the README (+ example). PTAL?